--- ext/dom/tests/dom005.phpt	2009-07-17 19:21:11.000000000 +0700
+++ ../php5.3-201004071230-old/ext/dom/tests/dom005.phpt	2010-04-08 01:15:20.000000000 +0700
@@ -27,10 +27,7 @@
 </body></html>
 --- save as HTML
 <!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.0 Transitional//EN" "http://www.w3.org/TR/REC-html40/loose.dtd">
-<html>
-<head><title>Hello world</title></head>
-<body>
+<html><head><title>Hello world</title></head><body>
 This is a not well-formed<br>
 html files with undeclared entities&nbsp;
-</body>
-</html>
+</body></html>
