diff -u a/ext/mysqli/mysqli.c b/ext/mysqli/mysqli.c
--- a/ext/mysqli/mysqli.c	2010-10-06 08:08:55.000000000 +0200
+++ b/ext/mysqli/mysqli.c	2010-12-08 21:42:15.000000000 +0100
@@ -543,6 +543,7 @@
 #else
 	STD_PHP_INI_ENTRY("mysqli.default_socket",			NULL,	PHP_INI_ALL,		OnUpdateStringUnempty,	default_socket,	zend_mysqli_globals,		mysqli_globals)
 #endif
+	STD_PHP_INI_ENTRY("mysqli.connect_charset",         NULL,   PHP_INI_ALL,        OnUpdateString,     connect_charset,    zend_mysqli_globals,        mysqli_globals)
 	STD_PHP_INI_BOOLEAN("mysqli.reconnect",				"0",	PHP_INI_SYSTEM,		OnUpdateLong,		reconnect,			zend_mysqli_globals,		mysqli_globals)
 	STD_PHP_INI_BOOLEAN("mysqli.allow_local_infile",	"1",	PHP_INI_SYSTEM,		OnUpdateLong,		allow_local_infile,	zend_mysqli_globals,		mysqli_globals)
 PHP_INI_END()
@@ -564,6 +565,7 @@
 	mysqli_globals->default_user = NULL;
 	mysqli_globals->default_pw = NULL;
 	mysqli_globals->default_socket = NULL;
+	mysqli_globals->connect_charset = NULL;
 	mysqli_globals->reconnect = 0;
 	mysqli_globals->report_mode = 0;
 	mysqli_globals->report_ht = 0;
Only in b/ext/mysqli: mysqli.c.orig
diff -u a/ext/mysqli/mysqli_nonapi.c b/ext/mysqli/mysqli_nonapi.c
--- a/ext/mysqli/mysqli_nonapi.c	2010-11-10 14:57:09.000000000 +0100
+++ b/ext/mysqli/mysqli_nonapi.c	2010-12-08 21:44:59.000000000 +0100
@@ -61,7 +61,7 @@
 	MY_MYSQL			*mysql = NULL;
 	MYSQLI_RESOURCE		*mysqli_resource = NULL;
 	zval				*object = getThis();
-	char				*hostname = NULL, *username=NULL, *passwd=NULL, *dbname=NULL, *socket=NULL;
+	char				*hostname = NULL, *username=NULL, *passwd=NULL, *dbname=NULL, *socket=NULL, *connect_charset=NULL;
 	int					hostname_len = 0, username_len = 0, passwd_len = 0, dbname_len = 0, socket_len = 0;
 	zend_bool			persistent = FALSE;
 	long				port = 0, flags = 0;
@@ -127,6 +127,11 @@
 	if (!socket_len || !socket) {
 		socket = MyG(default_socket);
 	}
+
+	connect_charset = MyG(connect_charset);
+	if (connect_charset != NULL) {
+		mysql_options(mysql->mysql, MYSQL_SET_CHARSET_NAME, connect_charset);
+	}
 	if (!port){
 		port = MyG(default_port);
 	}
Only in b/ext/mysqli: mysqli_nonapi.c.orig
diff -u a/ext/mysqli/php_mysqli_structs.h b/ext/mysqli/php_mysqli_structs.h
--- a/ext/mysqli/php_mysqli_structs.h	2010-10-04 12:02:58.000000000 +0200
+++ b/ext/mysqli/php_mysqli_structs.h	2010-12-08 21:42:15.000000000 +0100
@@ -291,6 +291,7 @@
 	char			*default_user;
 	char			*default_socket;
 	char			*default_pw;
+	char			*connect_charset;
 	long			reconnect;
 	long			allow_local_infile;
 	long			strict;
