diff -Naur ext/pdo_mysql//mysql_driver.c ext/pdo_mysql-patched//mysql_driver.c
--- ext/pdo_mysql//mysql_driver.c	2010-02-05 02:28:55.000000000 +0600
+++ ext/pdo_mysql-patched//mysql_driver.c	2010-03-05 16:56:15.861992835 +0600
@@ -565,6 +565,7 @@
 	pdo_mysql_db_handle *H;
 	int i, ret = 0;
 	char *host = NULL, *unix_socket = NULL;
+	char *connect_charset = NULL;
 	unsigned int port = 3306;
 	char *dbname;
 	struct pdo_data_src_parser vars[] = {
@@ -720,6 +721,11 @@
 		unix_socket = vars[4].optval;  
 	}
 
+	connect_charset = PDO_MYSQL_G(connect_charset);
+	if (connect_charset != NULL) {
+		mysql_options(H->server, MYSQL_SET_CHARSET_NAME, connect_charset);
+	}
+
 	/* TODO: - Check zval cache + ZTS */
 #ifdef PDO_USE_MYSQLND
 	if (dbname) {
diff -Naur ext/pdo_mysql//pdo_mysql.c ext/pdo_mysql-patched//pdo_mysql.c
--- ext/pdo_mysql//pdo_mysql.c	2010-02-04 15:37:38.000000000 +0600
+++ ext/pdo_mysql-patched//pdo_mysql.c	2010-03-05 17:02:30.031954897 +0600
@@ -56,6 +56,7 @@
 #if PDO_DBG_ENABLED
 	STD_PHP_INI_ENTRY("pdo_mysql.debug",	NULL, PHP_INI_SYSTEM, OnUpdateString, debug, zend_pdo_mysql_globals, pdo_mysql_globals)
 #endif
+	STD_PHP_INI_ENTRY("pdo_mysql.connect_charset",		NULL,	PHP_INI_ALL,	OnUpdateStringUnempty,		connect_charset,	zend_pdo_mysql_globals,		pdo_mysql_globals)
 PHP_INI_END()
 /* }}} */
 
@@ -165,6 +166,7 @@
 	pdo_mysql_globals->debug = NULL;	/* The actual string */
 	pdo_mysql_globals->dbg = NULL;	/* The DBG object*/
 #endif
+	pdo_mysql_globals->connect_charset = NULL;
 }
 /* }}} */
 #endif
diff -Naur ext/pdo_mysql//php_pdo_mysql_int.h ext/pdo_mysql-patched//php_pdo_mysql_int.h
--- ext/pdo_mysql//php_pdo_mysql_int.h	2010-02-04 15:37:38.000000000 +0600
+++ ext/pdo_mysql-patched//php_pdo_mysql_int.h	2010-03-05 17:04:32.315276092 +0600
@@ -75,6 +75,7 @@
 	 */
 	void *dummymemmber;
 #endif
+	char *connect_charset;
 ZEND_END_MODULE_GLOBALS(pdo_mysql)
 
 ZEND_EXTERN_MODULE_GLOBALS(pdo_mysql)
