diff -ur php-5.3.5.orig/scripts/phpize.in php-5.3.5/scripts/phpize.in
--- php-5.3.5.orig/scripts/phpize.in	2011-01-29 11:32:20.000000000 +0100
+++ php-5.3.5/scripts/phpize.in	2011-01-29 11:34:38.000000000 +0100
@@ -120,23 +120,6 @@
 {
   test -z "$PHP_AUTOCONF" && PHP_AUTOCONF=autoconf
   test -z "$PHP_AUTOHEADER" && PHP_AUTOHEADER=autoheader
-  
-  if test ! -x "$PHP_AUTOCONF" && test ! -x "`$php_shtool path $PHP_AUTOCONF`"; then
-    cat <<EOF
-Cannot find autoconf. Please check your autoconf installation and the
-\$PHP_AUTOCONF environment variable. Then, rerun this script.
-
-EOF
-    exit 1
-  fi
-  if test ! -x "$PHP_AUTOHEADER" && test ! -x "`$php_shtool path $PHP_AUTOHEADER`"; then
-    cat <<EOF
-Cannot find autoheader. Please check your autoconf installation and the
-\$PHP_AUTOHEADER environment variable. Then, rerun this script.
-
-EOF
-    exit 1
-  fi
 }
 
 phpize_copy_files()
